Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.lean

```diff
@@ -498,7 +499,7 @@ lemma negY_of_Z_ne_zero {P : Fin 3 → F} (hPz : P z ≠ 0) :
 lemma Y_sub_Y_mul_Y_sub_negY {P Q : Fin 3 → R} (hP : W'.Equation P) (hQ : W'.Equation Q)
     (hx : P x * Q z ^ 2 = Q x * P z ^ 2) :
     (P y * Q z ^ 3 - Q y * P z ^ 3) * (P y * Q z ^ 3 - W'.negY Q * P z ^ 3) = 0 := by
-  linear_combination (norm := (rw [negY]; ring1)) Q z ^ 6 * (equation_iff P).mp hP
+  linear_combination' (norm := (rw [negY]; ring1)) Q z ^ 6 * (equation_iff P).mp hP
     - P z ^ 6 * (equation_iff Q).mp hQ + hx * hx * hx + W'.a₂ * P z ^ 2 * Q z ^ 2 * hx * hx
     + (W'.a₄ * P z ^ 4 * Q z ^ 4 - W'.a₁ * P y * P z * Q z ^ 4) * hx
```

Comment:
The most direct translation would be to use the `congr(...)` term elaborator:
```lean
  linear_combination' (norm := (rw [negY]; ring1)) Q z ^ 6 * (equation_iff P).mp hP
    - P z ^ 6 * (equation_iff Q).mp hQ + congr($hx ^ 3) + W'.a₂ * P z ^ 2 * Q z ^ 2 * congr($hx ^ 2)
    + (W'.a₄ * P z ^ 4 * Q z ^ 4 - W'.a₁ * P y * P z * Q z ^ 4) * hx
```
But in general, a "nonlinear" linear combination has a code smell to me.  For example, I would find it more elegant to solve this example in this way, using a purely "linear" linear combination and avoiding the nonstandard discharger:
```lean
  rw [equation_iff, negY] at *
  linear_combination Q z ^ 6 * hP - P z ^ 6 * hQ
    + (Q z ^ 4 * P z ^ 4 * W'.a₄ - P y * Q z ^ 4 * P z * W'.a₁ 
      + Q z ^ 2 *  P z ^ 2 * (P z ^ 2 * Q x + Q z ^ 2 * P x) * W'.a₂ 
      + (P z ^ 4 * Q x ^ 2 + Q z ^ 2 * P z ^ 2 * Q x * P x + Q z ^ 4 * P x ^ 2)) * hx
```

Reply:
In any case, did you write these variants to check your understanding of the "story" of this PR, or are they intended as genuine suggested changes?  The approach I have taken on these two examples which are hit by the downgrade -- switching them over to `linear_combination'`, to serve as a record of the scope of functionality which has been removed --  was requested by @digama0:
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Narrowing.20the.20scope.20of.20.60linear_combination.60/near/465256620

Reply:
Thanks for the explanation, your snippet is indeed much more elegant!

So my suggestion is to just replace these two uses of `linear_combination'` with the solution you just wrote out, since thats nicer than the original anyways.

But also, I think the PR can be merged regardless if you do that or not :)

Reply:
If you don't mind either way, then let's go with the status quo, since Mario preferred it!